test/zaps/uni-v3/CloseLongZap.t.sol

```diff
@@ -0,0 +1,448 @@
+                // This will automatically be adjusted down.
+                amountIn: 885e18,
+                amountOutMinimum: 900e6
+            }),
+            false, // should wrap
+            false // as base
+        );
+    }
+
+    /// @dev Verify that `closeLongZap` performs correctly under the
+    ///      specified conditions.
+    /// @param _hyperdrive The Hyperdrive instance.
+    /// @param _maturityTime The maturity time of the long position.
+    /// @param _bondAmount The amount of longs to close.
+    /// @param _swapParams The Uniswap multi-hop swap parameters.
+    /// @param _shouldWrap A flag indicating whether or not the proceeds should
+    ///        be wrapped before the swap.
+    /// @param _asBase A flag indicating whether or not the deposit should be in
+    ///        base.
+    function _verifyCloseLongZap(
+        IHyperdrive _hyperdrive,
+        uint256 _maturityTime,
+        uint256 _bondAmount,
+        ISwapRouter.ExactInputParams memory _swapParams,
+        bool _shouldWrap,
+        bool _asBase
+    ) internal {
+        // Simulate closing the position without using the zap.
+        uint256 expectedHyperdriveVaultSharesBalanceAfter;
+        {
+            uint256 snapshotId = vm.snapshot();
+            _hyperdrive.closeLong(
+                _maturityTime,
+                _bondAmount,
+                0, // min output
+                IHyperdrive.Options({
+                    destination: alice,
+                    asBase: _asBase,
+                    extraData: new bytes(0)
+                })
+            );
+            expectedHyperdriveVaultSharesBalanceAfter = IERC20(
+                _hyperdrive.vaultSharesToken()
+            ).balanceOf(address(_hyperdrive));
+            vm.revertTo(snapshotId);
+        }
+
+        // Get some data before executing the zap.
+        uint256 longsOutstandingBefore = _hyperdrive
+            .getPoolInfo()
+            .longsOutstanding;
+        uint256 aliceOutputBalanceBefore;
+        address tokenOut = _swapParams.path.tokenOut();
+        if (tokenOut == WETH) {
+            aliceOutputBalanceBefore = alice.balance;
+        } else {
+            aliceOutputBalanceBefore = IERC20(tokenOut).balanceOf(alice);
+        }
+        uint256 aliceLongBalanceBefore = _hyperdrive.balanceOf(
+            AssetId.encodeAssetId(AssetId.AssetIdPrefix.Long, _maturityTime),
+            alice
+        );
+
+        // Execute the zap.
+        IHyperdrive hyperdrive = _hyperdrive; // avoid stack-too-deep
+        uint256 maturityTime = _maturityTime; // avoid stack-too-deep
+        uint256 bondAmount = _bondAmount; // avoid stack-too-deep
+        ISwapRouter.ExactInputParams memory swapParams = _swapParams; // avoid stack-too-deep
+        bool shouldWrap = _shouldWrap; // avoid stack-too-deep
+        bool asBase = _asBase; // avoid stack-too-deep
+        uint256 proceeds = zap.closeLongZap(
```

Comment:
weird that test coverage doesn't pick up on this

Reply:
Yeah it definitely is. Idk why